<launch>
  <arg name="ocs" default="true" />
  <remap from="/tf" to="/ocs/tf" if="$(arg ocs)"/>
  <remap from="/joint_states" to="/ocs/joint_states" if="$(arg ocs)"/>
  <remap from="/tf" to="/ocs/tf" if="$(arg ocs)"/>
  <remap from="/set_dynamic_tf" to="/ocs/set_dynamic_tf" if="$(arg ocs)"/>
  <remap from="/assoc_tf" to="/ocs/assoc_tf" if="$(arg ocs)"/>
  <remap from="/publish_tf" to="/ocs/publish_tf" if="$(arg ocs)"/>
  <remap from="/dissoc_tf" to="/ocs/dissoc_tf" if="$(arg ocs)"/>
  <remap from="/delete_tf" to="/ocs/delete_tf" if="$(arg ocs)"/>
  <remap from="/robot_description" to="/ocs/robot_description" />
  <arg name="INPUT_CLOUD" default="/ocs/communication/point_cloud" if="$(arg ocs)"/>
  <arg name="INPUT_CLOUD" default="/multisense/organized_image_points2_color" unless="$(arg ocs)"/>
  <arg name="default_frame_id" default="odom_on_ground" />
  <arg name="ik_server_launch" default="hrp2jsknts-ik-server.launch" />
  <arg name="robot_model_config" default="$(find drc_task_common)/config/robot_interactive_marker.yaml" />

  <include file="$(find drc_task_common)/launch/ocs/models/$(env ROBOT).launch" />

  <node pkg="drc_task_common" type="rviz_status.py" name="rviz_status">
  </node>
  <node pkg="jsk_pcl_ros" type="hinted_handle_estimator" name="hinted_handle_estimator" if="false">
    <remap from="~cloud" to="$(arg INPUT_CLOUD)" />
    <remap from="~point" to="/clicked_point" />
  </node>
  <!-- robot motion -->
  <include file="$(find drc_task_common)/launch/robot_motion.launch" >
    <arg name="default_frame_id" default="$(arg default_frame_id)" />
    <arg name="ik_server_launch" default="$(arg ik_server_launch)" />
    <arg name="ocs" default="$(arg ocs)" />
  </include>

  <!-- object model -->
  <include file="$(find drc_task_common)/launch/transformable_model.launch" >
    <arg name="default_frame_id" default="$(arg default_frame_id)" />
  </include>

  <!-- robot model -->
  <include file="$(find drc_task_common)/launch/robot_model.launch" >
    <arg name="default_frame_id" default="$(arg default_frame_id)" />
    <arg name="robot_model_config" default="$(arg robot_model_config)" />
  </include>

  <node pkg="drc_task_common" type="joint-states-storage.l" name="joint_states_storage"
        />
  <!-- teleop device -->
  <include file="$(find drc_task_common)/launch/teleop_device.launch" >
    <arg name="default_frame_id" default="$(arg default_frame_id)" />
    <arg name="enable_trackball_head" default="false" />
  </include>

  <!-- teleop interface -->
  <include file="$(find drc_task_common)/launch/teleop_interface.launch" />

  <!-- stair marker -->
  <node pkg="drc_task_common" type="stair_marker.py" name="stair_marker" />
  <!-- marker -->
  <node pkg="jsk_interactive_marker"
        type="footstep_marker"
        name="footstep_marker"
        output="screen"
        if="false"
        >
    <remap from="/lookaround_ground" to="terrain_lookground" />
    <remap from="~projected_pose" to="projected_footstep_ocs" />
    <!-- <remap from="~grid_arrays" to="/locomotion/env_server/output" /> -->
    <!-- <remap from="~planes_coefficients" to="/locomotion/plane_concatenator/output/coefficients" /> -->
    <!-- <remap from="move_marker" to="$(arg POSE_REMAP)" /> -->
    <!-- <param name="wait_snapit_server" value="false" /> -->
    <rosparam command="load" file="$(find jsk_footstep_controller)/config/jaxon.yaml" />
    <remap from="project_footprint" to="~project_footprint" />
    <remap from="footstep_planner" to="ocs_footstep_planner" />
    <remap from="footstep_controller" to="ocs_footstep_controller" />
    <rosparam>
      use_plane_snap: true
      use_projection_service: false
      use_projection_topic: true
      use_footstep_planner: true
      use_footstep_controller: true
      use_initial_footstep_tf: true
      initial_reference_frame: '/ground'
      # HRP2JSK
      foot_size_x: 0.235
      lfoot_offset: [0.02, 0.01, -0.105, 0, 0, 0, 1]
      rfoot_offset: [0.02, -0.01, -0.105, 0, 0, 0, 1]
      
      foot_size_z: 0.01
    </rosparam>
    <param name="frame_id" value="/ground" />
  </node>
  <node pkg="jsk_footstep_controller" type="footstep-overlay-text.l"
        name="footstep_overlay_text">
  </node>
  <!-- rviz -->
  <node pkg="rviz" type="rviz" name="operator_rviz" args="-d $(find drc_task_common)/config/drc_task_common.rviz"
        respawn="true">
    <rosparam command="load" file="$(find drc_task_common)/config/ocs_rviz_buttons.yaml" />
  </node>
</launch>
